fingertip/plugins/script.py

```diff
@@ -31,7 +35,36 @@ def run(m, scriptpath, cache=0, no_unseal=False):
         m.ssh.upload(scriptpath, dst=scriptname)
         m.log.debug(f'uploaded {scriptpath} into {scriptname}')
         m(f'chmod +x {scriptname}')
+        if mode == 'test':
+            m.hooks.disable_proxy()
+            check = False
+        elif mode == 'run':
+            check = True
+        else:
+            assert False
         m.log.plain()
-        m(f'./{scriptname}')
+        exit_code = m(f'./{scriptname}', check=check).retcode
         m.log.nicer()
+        if mode == 'test' and exit_code != 0:
```

Comment:
same as for `disable_proxy`, may be easier to return both machine and retcode, decide what to do with them outside and maybe turn `keep _exec` mode-less.

Reply:
I don't understand "turn `keep _exec` mode-less" part. But don't want to redo if there is no clear benefit at this moment.

Reply:
I am starting to understand maybe.
Thinking more about `disable_proxy` actually, in current form, we are also disabling the proxy for consecutive plugins in the pipeline, right? Not sure if it's desirable or not though.